### DISCUSSION TITLE
Global spring configuration

### DISCUSSION THREAD

**Discussion by Emiliano-Bucci** [pmndrs/react-spring#1428] [Support]
Hi! I was looking into the `Globals` but I couldn't find anything related: is there a way to set a global spring configuration without having to pass it to every useSpring, set, ecc call every time? I guess it would be useful to do that

**Comment by joshuaellis** [comment]
Like a default config?

**Reply by Emiliano-Bucci** [reply]
Yes, exactly

**Reply by joshuaellis** [reply]
Have you looked at `SpringContext`? It's a new component – https://aleclarson.github.io/react-spring/v9/#The-SpringContext-component

**Reply by Emiliano-Bucci** [reply]
@joshuaellis I didn't know it! But for some reason I cannot add a config prop. I'm using `react-spring v^9.1.1`

![image](https://user-images.githubusercontent.com/36669369/115840999-9faeba80-a41c-11eb-8d76-d06406168ee5.png)

Do you know what could be? Or am I doing something wrong?

EDIT: It seems only a type issue, I can actually pass the prop and it works. Maybe I can create a small PR? :)

**Reply by joshuaellis** [reply]
Absolutely! Please do, I can look closer before we merge too 🤓 

If you feel like this was the correct answer, please tick it to show that ⭐

**Reply by Emiliano-Bucci** [reply]
@joshuaellis I was investigating further, and i've realized that it isn't working, somehow. Now, probably there're lots of things I'm missing here, but as far i can see from the logs, i see that the value passed to the `<SpringContext />` are only used to populate the `defaultProps`:

![image](https://user-images.githubusercontent.com/36669369/115892784-c9390780-a457-11eb-8bb4-62df3b7f8f62.png)

This is a log example of a simple spring i have: the `defaultProps` are populated with the config passed to the `<SpringContext />`, but the `animation config` currently used to animate is the one in the `opacity.animation.config`, which is the default one used by `react-soring` under the hood (that can be override by passing a custom config to a single spring hook).

**Reply by joshuaellis** [reply]
AFAIK the default props are merged with the props you pass. I'm not sure if it's deep merge so if you define a config, is that merged too or simply overwritten? I'd assume it's merged otherwise your spring might have mass but no tension?

Looking in the source would give you some answers I reckon.

**Reply by Emiliano-Bucci** [reply]
Thanks for the info; i still need to investigate better and understand better the flow :)
